File: backend/src/main/java/com/lsiccha/procesos/services/ProcesoService.java
package com.lsiccha.procesos.services;

import com.lsiccha.procesos.model.Diagrama;
import com.lsiccha.procesos.model.Proceso;
import com.lsiccha.procesos.model.User;
import org.springframework.stereotype.Service;

import java.util.List;

public interface ProcesoService {

    Proceso create(Proceso proceso);
    List<Proceso> getAll();
    Proceso get(Integer id);
    Proceso update(Proceso proceso);
    void delete(Integer id);

    List<Proceso> getByDiagrama(Diagrama diagrama);

}

File: diagrama-procesos/src/app/services/proceso.service.ts
import { Injectable } from '@angular/core';
import {HttpClient, HttpParams} from "@angular/common/http";
import {Process} from "../model/process.model";
import {environment} from "../../environments/environment";
import {Diagrama} from "../model/diagrama.model";
import {Observable} from "rxjs";

@Injectable({
  providedIn: 'root'
})
export class ProcesoService {

  private apiBase : string = `${environment.apiBase}/procesos`

  constructor(
    private http: HttpClient
  ) { }

  create(proceso: Process){
    return this.http.post(
      this.apiBase,
      proceso
    )
  }

  getAll(){
    return this.http.get(
      this.apiBase
    )
  }

  get(id: number){
    return this.http.get(
      `${this.apiBase}/${id}`
    )
  }

  update(proceso: Process){
    return this.http.put(
      this.apiBase,
      proceso
    )
  }

  delete(id: number){
    return this.http.delete(
      `${this.apiBase}/${id}`
    )
  }

  getByDiagrama(diagrama: number) : Observable<Process[]>{
    let params = new HttpParams()
    params = params.set('diagrama', diagrama);
    return this.http.get<Process[]>(
      `${this.apiBase}/byDiagrama`,
      {
        params : params
      }
    )
  }



}
